netdata/netdata
# Fix chart update ebpf.plugin

##### Summary
Fixes #13346 

This PR is bringing the following changes to remove locks that do not allow chart to be updated correctly:

1. Change `else` to `else if` in `systemd` charts, this will avoid to recreate these charts every iteration
2. Use secondary mutex only when thread is integrated with `apps' and `cgroup` will work more independent. 
3. Remove calls to functions `broadcast` and `wait`, because they were creating delay to send the messages to netdata on environments tested

##### Test Plan

1. Compile current master branch and run netdata with `ebpf.plugin` enabled during few minutes
2. Run the following command and verify that you will have value different than `0`:

```sh
$ grep rrdset_done /var/log/netdata/error.log  | grep ebpf | wc -l
456
```
3. Stop netdata
4. Compile this branch
5. Run step 2 again:
```sh
$ grep rrdset_done /var/log/netdata/error.log  | grep ebpf | wc -l
0
```
##### Additional Information
This PR was tested on the following environments:

| database | branch |result |
|----------|--------|-------|
|map       | master | Errors appear |
|dbengine  | master | error appear |
|dbengine  | this branch | After this [commit](https://github.com/netdata/netdata/pull/13351/commits/ee33b19c2c9a25f1d558dd666db0ab1efd18f9fe) with all threads not related to apps/cgroup disabled, error did not appear |
|dbengine  | this branch | All threads enabled, I had only 9 errors, but this happened when parent was overload, and child was a VM |
|dbengine with 2 Tier | this Branch |  like previous |
|map       | this branch | All threads enabled. | Like the previous, but I observed a high frequency of errors when parent was overloaded.
|ram       | this branch | All threads enabled. When I ran on parent I did not have any error. |

Map is the memory mode where I had more events, and I still have when my parent was overloaded or updating Tier.

<details> <summary>For users: How does this change affect me?</summary>
Describe the PR affects users: 
- Which area of Netdata is affected by the change? ebpf.plugin
- Can they see the change or is it an under the hood? If they can see it, where? Due the fact netdata interpolates, it was not visible when a data was not stored on dashboard, but this is visible trough our API.
- How is the user impacted by the change? All collected data will be stored in the correct time, unless system is overloaded, but this also affects other plugins.
- What are there any benefits of the change? Data quality
</details>
